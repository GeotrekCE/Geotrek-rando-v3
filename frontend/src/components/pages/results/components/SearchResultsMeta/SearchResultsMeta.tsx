import React from 'react';
import styled, { css } from 'styled-components';
import { colorPalette, desktopOnly, getSpacing, typography } from 'stylesheet';

export const SearchResultsMeta: React.FC = () => {
  return (
    <div className="flex">
      <div>
        <Illustration className="hidden desktop:block" src="images/little-forest.png" />
      </div>

      <div className="desktop:ml-6">
        <ResultsNumber>82 résultats trouvés</ResultsNumber>
        <RankingInfo className="desktop:hidden">Classement par ordre de pertinence</RankingInfo>
      </div>
    </div>
  );
};

const Illustration = styled.img`
  height: ${getSpacing(16)};
  width: ${getSpacing(16)};
`;

const ResultsNumber = styled.div`
  ${typography.h1};
  color: ${colorPalette.darkPurple};

  ${desktopOnly(css`
    ${typography.h2}
  `)}
`;

const RankingInfo = styled.div`
  ${typography.small}
  margin-top: ${getSpacing(1)};
`;
