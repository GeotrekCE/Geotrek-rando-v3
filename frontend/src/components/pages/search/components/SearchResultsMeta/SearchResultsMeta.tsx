import React from 'react';
import styled, { css } from 'styled-components';
import { FormattedMessage } from 'react-intl';
import { colorPalette, desktopOnly, getSpacing, typography } from 'stylesheet';

interface Props {
  resultsNumber: number | undefined;
}

export const SearchResultsMeta: React.FC<Props> = ({ resultsNumber }) => {
  return (
    <div className="flex">
      <div>
        <Illustration className="hidden desktop:block" src="/images/little-forest.png" />
      </div>

      <div className="desktop:ml-6">
        <ResultsNumber>
          <FormattedMessage values={{ count: resultsNumber }} id="search.resultsFound" />
        </ResultsNumber>
      </div>
    </div>
  );
};

const Illustration = styled.img`
  height: ${getSpacing(16)};
  width: ${getSpacing(16)};
`;

const ResultsNumber = styled.div`
  ${typography.h3};
  color: ${colorPalette.darkPurple};

  ${desktopOnly(css`
    ${typography.h2}
  `)}
`;
