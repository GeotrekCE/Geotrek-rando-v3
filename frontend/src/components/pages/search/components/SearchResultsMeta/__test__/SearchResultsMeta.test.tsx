import { render } from 'services/testing/reactTestingLibraryWrapper';

import { SearchResultsMeta } from '../SearchResultsMeta';

describe('AAU, I can see informations on my search', () => {
  it('with 0 results', () => {
    const component = render(
      <SearchResultsMeta resultsNumber={0} placeName="Paris" placeUrl="/" />,
    );
    expect(component).toMatchSnapshot();
  });

  it('with 1 result', () => {
    const component = render(
      <SearchResultsMeta resultsNumber={1} placeName="Paris" placeUrl="/" />,
    );
    expect(component).toMatchSnapshot();
  });

  it('with many results', () => {
    const component = render(
      <SearchResultsMeta resultsNumber={123} placeName="Paris" placeUrl="/" />,
    );
    expect(component).toMatchSnapshot();
  });
});
