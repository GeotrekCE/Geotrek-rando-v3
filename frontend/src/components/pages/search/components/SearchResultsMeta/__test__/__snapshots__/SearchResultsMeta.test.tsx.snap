// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AAU, I can see informations on my search with 0 results 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex"
      >
        <div>
          <img
            class="SearchResultsMeta__Illustration-sc-18aa4n-0 bxnLjy hidden desktop:block"
            src="/images/little-forest.png"
          />
        </div>
        <div
          class="desktop:ml-6"
        >
          <div
            class="SearchResultsMeta__ResultsNumber-sc-18aa4n-1 fYecrv"
          >
            0 résultat trouvé
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex"
    >
      <div>
        <img
          class="SearchResultsMeta__Illustration-sc-18aa4n-0 bxnLjy hidden desktop:block"
          src="/images/little-forest.png"
        />
      </div>
      <div
        class="desktop:ml-6"
      >
        <div
          class="SearchResultsMeta__ResultsNumber-sc-18aa4n-1 fYecrv"
        >
          0 résultat trouvé
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`AAU, I can see informations on my search with 1 result 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex"
      >
        <div>
          <img
            class="SearchResultsMeta__Illustration-sc-18aa4n-0 bxnLjy hidden desktop:block"
            src="/images/little-forest.png"
          />
        </div>
        <div
          class="desktop:ml-6"
        >
          <div
            class="SearchResultsMeta__ResultsNumber-sc-18aa4n-1 fYecrv"
          >
            1 résultat trouvé
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex"
    >
      <div>
        <img
          class="SearchResultsMeta__Illustration-sc-18aa4n-0 bxnLjy hidden desktop:block"
          src="/images/little-forest.png"
        />
      </div>
      <div
        class="desktop:ml-6"
      >
        <div
          class="SearchResultsMeta__ResultsNumber-sc-18aa4n-1 fYecrv"
        >
          1 résultat trouvé
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`AAU, I can see informations on my search with many results 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex"
      >
        <div>
          <img
            class="SearchResultsMeta__Illustration-sc-18aa4n-0 bxnLjy hidden desktop:block"
            src="/images/little-forest.png"
          />
        </div>
        <div
          class="desktop:ml-6"
        >
          <div
            class="SearchResultsMeta__ResultsNumber-sc-18aa4n-1 fYecrv"
          >
            123 résultats trouvés
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex"
    >
      <div>
        <img
          class="SearchResultsMeta__Illustration-sc-18aa4n-0 bxnLjy hidden desktop:block"
          src="/images/little-forest.png"
        />
      </div>
      <div
        class="desktop:ml-6"
      >
        <div
          class="SearchResultsMeta__ResultsNumber-sc-18aa4n-1 fYecrv"
        >
          123 résultats trouvés
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
